src/components/callout/callout.tsx

```diff
@@ -0,0 +1,86 @@
+
+/**
+ * Callouts –also known as Admonitions– are useful for including supportive
+ * or special content within a large piece of text,
+ * or even inside a user interface.
+ *
+ * When used in a document or text based user interface, the callout attract
+ * reader's attention to a particular piece of information,
+ * without significantly interrupting their flow of reading the document.
+ *
+ * On a user interface, their usage will be experienced more intrusively by
+ * end-users. But they could be good choices when you indent to slightly
+ * disturb users' attention, and challenge them to pay extra attention
+ * to the information presented. In such cases, callouts should not
+ * be used as static and constantly present elements of a UI. They should
+ * rather be displayed when something unusual or remarkable demands
+ * users attention.
+ *
+ * @exampleComponent limel-example-callout-info
+ * @exampleComponent limel-example-callout-warning
+ * @exampleComponent limel-example-callout-danger
+ * @exampleComponent limel-example-callout-note
+ * @exampleComponent limel-example-callout-tip
+ * @exampleComponent limel-example-callout-success
+ * @exampleComponent limel-example-callout-failure
+ * @exampleComponent limel-example-callout-example
+ * @exampleComponent limel-example-callout-quote
+ * @exampleComponent limel-example-callout-question
+ * @exampleComponent limel-example-callout-custom-heading
+ * @exampleComponent limel-example-callout-custom-icon
+ * @exampleComponent limel-example-callout-custom-component
+ * @exampleComponent limel-example-callout-styles
+ * @exampleComponent limel-example-callout-composite
+ */
+@Component({
+    tag: 'limel-callout',
+    shadow: true,
+    styleUrl: 'callout.scss',
+})
+export class Callout {
+    /**
+     * Heading of the callout, which can be used to override the
+     * default heading which is displayed based on the chosen `type`.
+     */
+    @Prop({ reflect: true })
+    public heading?: string;
+
+    /**
+     * Icon of the callout, which can be used to override the
+     * default icon which is displayed based on the chosen `type`.
+     */
+    @Prop({ reflect: true })
+    public icon?: string;
+
+    /**
+     * Defines how the component is visualized, for example
+     * which heading, color or icon is used in its user interface.
+     */
+    @Prop({ reflect: true })
+    public type?: CalloutType = 'info';
+
+    /**
+     * Defines the language for translations.
+     * Will translate the default headings for supported languages.
+     */
+    @Prop()
+    public language: Languages = 'en';
+
+    public render() {
+        return [
+            <div class="side" role="presentation">
+                <limel-icon name={getIcon(this.icon, this.type)} />
+            </div>,
+            <div class="main">
+                <h1 class="heading">
+                    {getHeading(this.heading, this.type, this.language)}
+                </h1>
+                <slot name="content" />
```

Comment:
Is there a reason to why we named the slot? Since there is only one slot it will be easier to use if the consumer don't have to add the `slot` attribute to their tags.

Reply:
I agree that an unnamed slot is much easier to use, @Kiarokh said that there was some kind of discussion regarding this that led to that all slots should be named in lime-elements.

But for me it doesn't matter.

Reply:
Will it work to do something like this? Otherwise, you would be forced to add an inner tag as well just to specify the slot which is not ideal
```html
<limel-callout>Hello, world!</limel-callout>
```

Reply:
https://github.com/Lundalogik/lime-elements/pull/2167#discussion_r1147762984

Reply:
> Or we don't name the main slot (the content slot), but name every other slot.

If there's more than one slot, one of them needs to be named or it won't work

For all cases where it's quite obvious what the main content is (like in this component), I see no reason to why we need to name the slot

Reply:
I'm fine with not naming this slot, as it seems pretty obvious that if more slots are ever added, this will remain the main one. The problem was in some other case, where we had only one slot, but that slot was for action buttons or something, and then, later, we wanted to add another slot for something like the content of the "body", and all of a sudden we had a situation where the default slot was very much an aside kind of thing, while the clearly main slot was named 🙈

> Will it work to do something like this? Otherwise, you would be forced to add an inner tag as well just to specify the slot which is not ideal
> 
> ```
> <limel-callout>Hello, world!</limel-callout>
> ```

I think this argument is powerful. Let's remove the name 👍

I'll fix it, since I'll remove some of the extra types anyway.

Reply:
:a:

Reply:
⚡